### PR TITLE
Adds phone auth testing. (disableAppVerification)

### PR DESCRIPTION
**Pull request by protocol86**
Adds API for phone auth testing.

Enabling this flag essentially bypasses app verification during the "verify phone number" flow.

### CODE REVIEW

File: Example/Auth/Tests/FIRPhoneAuthProviderTests.m [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #import "FIRAuthUIDelegate.h"
 #import "FIRAuthURLPresenter.h"
 #import "FIRAuthWebUtils.h"
+#import "FIRAuthSettings.h"
```
Comment (ryanwilson) [inline Example/Auth/Tests/FIRPhoneAuthProviderTests.m]: Nit: Ordering, the rest of the headers are alphabetical.

File: Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m [hunk 2]
```diff
@@ -364,6 +365,20 @@ - (void)internalVerifyPhoneNumber:(NSString *)phoneNumber
 - (void)verifyClientAndSendVerificationCodeToPhoneNumber:(NSString *)phoneNumber
                              retryOnInvalidAppCredential:(BOOL)retryOnInvalidAppCredential
                                                 callback:(FIRVerificationResultCallback)callback {
+  if (_auth.settings.isAppVerificationDisabledForTesting) {
+    FIRSendVerificationCodeRequest *request =
+        [[FIRSendVerificationCodeRequest alloc] initWithPhoneNumber:phoneNumber
+                                                     appCredential:nil
+                                                    reCAPTCHAToken:nil
+                                              requestConfiguration:
+                                                  self->_auth.requestConfiguration];
```
Comment (ryanwilson) [inline Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: Nit: Is there a reason you use `self->` here but not above in `_auth.settings. isAppVerificationDisabledForTesting`?

File: Firebase/Auth/Source/FIRAuthSettings.m [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 Google
```
Comment (ryanwilson) [inline Firebase/Auth/Source/FIRAuthSettings.m]: s/2017/2018

File: Firebase/Auth/Source/Public/FIRAuth.h [hunk 4]
```diff
@@ -27,6 +27,7 @@
 @class FIRApp;
 @class FIRAuth;
 @class FIRAuthCredential;
+@class FIRAuthSettings;
```
Comment (ryanwilson) [inline Firebase/Auth/Source/Public/FIRAuth.h]: Nit: ordering, alphabetical.

File: Firebase/Auth/Source/Public/FIRAuthSettings.h [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017 Google
```
Comment (ryanwilson) [inline Firebase/Auth/Source/Public/FIRAuthSettings.h]: s/2017/2018

File: Firebase/Auth/Source/RPCs/FIRAuthBackend.m [hunk 6]
```diff
@@ -1069,7 +1076,8 @@ + (nullable NSError *)clientErrorWithServerErrorMessage:(NSString *)serverErrorM
     return [FIRAuthErrorUtils quotaExceededErrorWithMessage:serverErrorMessage];
   }
 
-  if ([shortErrorMessage isEqualToString:kAppNotVerifiedErrorMessage]) {
+  if ([shortErrorMessage isEqualToString:kAppNotVerifiedErrorMessage] ||
```
Comment (ryanwilson) [inline Firebase/Auth/Source/RPCs/FIRAuthBackend.m]: Just to confirm, we want to return an error stating the app isn't verified in the situation where they insert a real phone number but testing is enabled? Wouldn't it make sense to have a specific error for this?

**Comment by paulb777** [comment]
@ryanwilson Auth is not currently compatible with running `style.sh`

@protocol86 LGTM after @ryanwilson's review is resolved.

File: Example/Auth/Tests/FIRPhoneAuthProviderTests.m (hunk 1 above)
Comment (protocol86) [reply Example/Auth/Tests/FIRPhoneAuthProviderTests.m]: Done, thanks.

File: Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m (hunk 2 above)
Comment (protocol86) [reply Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: Good catch, removed it to improve readability.

File: Firebase/Auth/Source/FIRAuthSettings.m (hunk 3 above)
Comment (protocol86) [reply Firebase/Auth/Source/FIRAuthSettings.m]: Done.

File: Firebase/Auth/Source/Public/FIRAuth.h (hunk 4 above)
Comment (protocol86) [reply Firebase/Auth/Source/Public/FIRAuth.h]: Done.

File: Firebase/Auth/Source/Public/FIRAuthSettings.h (hunk 5 above)
Comment (protocol86) [reply Firebase/Auth/Source/Public/FIRAuthSettings.h]: Done.

File: Firebase/Auth/Source/RPCs/FIRAuthBackend.m (hunk 6 above)
Comment (protocol86) [reply Firebase/Auth/Source/RPCs/FIRAuthBackend.m]: Changed the error message for this specific case.

**Pull request merged**